### ISSUE TITLE
Blocks not allowed for a post type are shown in the Pattern directory

### ISSUE THREAD

**Issue by chvillanuevap** [WordPress/gutenberg#24505]
**Describe the bug**
Blocks types that are not allowed using the `allowed_block_types` filters are still shown in the Pattern directory.
**To reproduce**
1. Select which blocks are allowed using the `allowed_block_types` filter.

```
add_filter( 'allowed_block_types', array( 'my_allowed_block_types' ), PHP_INT_MAX, 2 );
function my_allowed_block_types( $allowed_blocks, $post ) {
    if ( in_array( $post->post_type, [ 'mycpt' ], true ) ) {
        return [
            ... // my allowed block types
        ];
    }

    return $allowed_blocks;
}
```
2. Click on the appender "+" button in your post type.
3. Select the Patterns tab.
4. See error.

**Expected behavior**
Since I'm only allowing custom blocks that I built myself in my post type, the pattern directory should be empty because I haven't added any patterns that use my block.

**Editor version (please complete the following information):**
- WordPress version: 5.5
- Does the website has Gutenberg plugin installed, or is it using the block editor that comes by default? Default.

**Comment by jeremyfelt** [comment]
Related: #23275

**Comment by jeremyfelt** [comment]
This was partially addressed in #30300. I think this ticket (#24505) can be closed as a duplicate of #23275.

**Comment by Mamaduka** [comment]
I think this should be fixed after #30366.

**Comment by youknowriad** [comment]
Closing as the linked PR is merged.
